sponsorship_compassion/model/contracts.py

```diff
@@ -61,8 +67,52 @@ def _get_contract_from_invoice(invoice_obj, cr, uid, invoice_ids,
                         contract.write({
                             'activation_date': datetime.today().strftime(DF)})
 
+                        # Cancel the old invoices if a contract is activated
+                        self._cancel_old_invoices(
+                            cr, uid,
+                            invoice.partner_id.id,
+                            contract.id,
+                            first_pay_date,
+                            context)
         return list(res)
 
+    def _cancel_old_invoices(
+            self, cr, uid, partner_id,
+            contract_id, date_invoice, context=None):
+        invoice_obj = self.pool.get('account.invoice')
+        invoice_ids = invoice_obj.search(
+            cr, uid,
+            [('partner_id', '=', partner_id),
+             ('state', '=', 'open'),
+             ('date_invoice', '<', date_invoice)
+             ],
+            context=context)
+
+        for invoice in invoice_obj.browse(cr, uid, invoice_ids, context):
+            invoice_lines = invoice.invoice_line
+            contract_ids = [
+                invoice_line.contract_id.id for invoice_line in invoice_lines

```

Comment:
you could rename invoice_line to invl inside the brackets because it is just used here, and it could save a bit of line length. But if you prefer like this, you can keep it.